cmd/kubeadm/app/cmd/reset.go

```diff
@@ -206,9 +231,17 @@ func newCmdReset(in io.Reader, out io.Writer, resetOptions *resetOptions) *cobra
 		if cmd.Flags().Lookup(options.NodeCRISocket) == nil {
 			// avoid CRI detection
 			// assume that the command execution does not depend on CRISocket when --cri-socket flag is not set
-			resetOptions.criSocketPath = kubeadmconstants.UnknownCRISocket
+			resetOptions.externalcfg.CRISocket = kubeadmconstants.UnknownCRISocket
 		}
-		return newResetData(cmd, resetOptions, in, out)
+		data, err := newResetData(cmd, resetOptions, in, out, true)
+		if err != nil {
+			return nil, err
+		}
+		// If the flag for skipping phases was empty, use the values from config
+		if len(resetRunner.Options.SkipPhases) == 0 {
+			resetRunner.Options.SkipPhases = data.resetCfg.SkipPhases
```

Comment:
- `SkipPhases` will take from flag first and then configuration. No merge.
- `ignorePreflightErrors` will try to merge flag set with configuration ignorePreflightErrors set.

Reply:
haven't checked the overrides more closely, but i think if the CLI allows a `[]` and the config allows a `[]`, then the CLI `[]` should simply replace the config `[]`. there should not be a merge / append.

i don't know if we are doing it like that everywhere, but it makes sense to reduce user confusion - flags "override" config.

Reply:
> SkipPhases will take from the flag first and then configuration. No merge.
> ignorePreflightErrors will try to merge the flag set with the configuration ignorePreflightErrors set.

true,  all what have done here for the `SkipPhases` and `ignorePreflightErrors` is trying to follow the existing pattern with init/join. It does not consistent for the time being, update this for init/join will cause some behavior change if we make it simply override the config in the v1beta4, to be decided.